modules/siddhi-extensions/markov-models/src/main/java/org/wso2/siddhi/extension/markov/MarkovChainStreamProcessor.java

```diff
@@ -0,0 +1,280 @@
+            }
+
+        } else if (object instanceof Integer) {
+            notificationsHoldLimit = (Integer) object;
+        } else if (object instanceof Long) {
+            notificationsHoldLimit = (Long) object;
+        } else {
+            throw new ExecutionPlanValidationException(
+                    "5th parameter should be the Training batch size or Markov matrix storage location. "
+                            + "They should be of types int/long or String. But found "
+                            + attributeExpressionExecutors[4].getReturnType());
+        }
+
+        if (attributeExpressionExecutors.length == 6) {
+
+            if (attributeExpressionExecutors[5] instanceof ConstantExpressionExecutor) {
+                Object trainingOptionObject = attributeExpressionExecutors[5].execute(null);
+                if (trainingOptionObject instanceof Boolean) {
+                    trainingOption = (Boolean) trainingOptionObject;
+                } else {
+                    throw new ExecutionPlanValidationException("Training option should be of type boolean. But found "
+                            + attributeExpressionExecutors[5].getReturnType());
+                }
+            } else {
+                trainingOptionExpressionExecutor = attributeExpressionExecutors[5];
+            }
+
+        }
+
+        if (trainingMode == TrainingMode.PREDEFINED_MATRIX) {
+            markovChainTransitionProbabilitiesCalculator = new MarkovChainTransitionProbabilitiesCalculator(
+                    durationToKeep, alertThresholdProbability, markovMatrixStorageLocation);
+        } else {
+            markovChainTransitionProbabilitiesCalculator = new MarkovChainTransitionProbabilitiesCalculator(
+                    durationToKeep, alertThresholdProbability, notificationsHoldLimit);
+
+        }
+
+        List<Attribute> attributeList = new ArrayList<Attribute>(3);
+        attributeList.add(new Attribute("lastState", Attribute.Type.STRING));
+        attributeList.add(new Attribute("transitionProbability", Attribute.Type.DOUBLE));
+        attributeList.add(new Attribute("notify", Attribute.Type.BOOL));
+        return attributeList;
+    }
+
+    /**
+     * The main processing method that will be called upon event arrival
+     *
+     * @param streamEventChunk the event chunk that need to be processed
+     * @param nextProcessor the next processor to which the success events need to be passed
+     * @param streamEventCloner helps to clone the incoming event for local storage or
+     *            modification
+     * @param complexEventPopulater helps to populate the events with the resultant attributes
+     */
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+            StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater) {
+
+        if (streamEventChunk.getFirst() == null) {
+            return;
+        }
```

Comment:
Thanks for pointing it out. Seems like practically that cannot happen. Fixed with https://github.com/wso2/siddhi/pull/189/commits/690b60924a459f505cce35f57d2ce6b05e4980c2